query/math.go

```diff
@@ -42,13 +42,70 @@ func (v *MathValuer) Value(key string) (interface{}, bool) {
 
 func (v *MathValuer) Call(name string, args []influxql.Expr) (interface{}, bool) {
 	if len(args) == 1 {
+		arg0, ok := v.getFloatValue(args[0])
+		if !ok {
+			return nil, false
+		}
+		
+		var fn func(float64) float64
 		switch name {
+		case "abs":
+			fn = math.Abs
 		case "sin":
-			return v.callTrigFunction(math.Sin, args[0])
+			fn =  math.Sin
 		case "cos":
-			return v.callTrigFunction(math.Cos, args[0])
+			fn =  math.Cos
 		case "tan":
-			return v.callTrigFunction(math.Tan, args[0])
+			fn =  math.Tan
+		case "asin":
+			fn =  math.Asin
+		case "acos":
+			fn =  math.Acos
+		case "atan":
+			fn =  math.Atan
+		case "exp":
+			fn =  math.Exp
+		case "log":
+			fn =  math.Log
+		case "log10":
+			fn =  math.Log10
+		case "sqrt":
+			fn =  math.Sqrt
+		}
+
+		if fn != nil {
+			result := fn(arg0)
+			if math.IsNaN(result) {
+				return nil, false
+			} else {
+				return result, true
+			}
+		}
+	} else if len(args) == 2 {
+		arg0, ok := v.getFloatValue(args[0])
+		if !ok {
+			return nil, false
+		}
+		arg1, ok := v.getFloatValue(args[1])
+		if !ok {
+			return nil, false
+		}
+		
+		var fn func(float64, float64) float64
+		switch name {
+		case "atan2":
+			fn = math.Atan2
+		case "pow":
+			fn = math.Pow
+		}
+		
+		if fn != nil {
+			result := fn(arg0, arg1)
+			if math.IsNaN(result) {
+				return nil, false
```

Comment:
This should probably return `nil, true` and should probably be more like this:

```golang
result := fn(arg0, arg1)
if math.IsNaN(result) {
    result = nil
}
return result, true
```

The `ok` is supposed to signify "was this call handled?" It was handled, it just returned a result that we do not support and cannot serialize back to the user to begin with.

Reply:
That makes sense, I guessed that might be what I should return but wasn't sure.